pwndbg/gdblib/vmmap.py

```diff
@@ -251,25 +252,7 @@ def coredump_maps() -> Tuple[pwndbg.lib.memory.Page, ...]:
     Parses `info proc mappings` and `maintenance info sections`
     and tries to make sense out of the result :)
     """
-    pages: List[pwndbg.lib.memory.Page] = []
-
-    try:
-        info_proc_mappings = pwndbg.gdblib.info.proc_mappings().splitlines()
-    except gdb.error:
-        # On qemu user emulation, we may get: gdb.error: Not supported on this target.
-        info_proc_mappings = []
-
-    for line in info_proc_mappings:
-        # We look for lines like:
-        # ['0x555555555000', '0x555555556000', '0x1000', '0x1000', '/home/user/a.out']
-        try:
-            start, _end, size, offset, objfile = line.split()
-            start, size, offset = int(start, 16), int(size, 16), int(offset, 16)
-        except (IndexError, ValueError):
-            continue
-
-        # Note: we set flags=0 because we do not have this information here
-        pages.append(pwndbg.lib.memory.Page(start, size, 0, offset, objfile))
+    pages = list(info_proc_maps())
```

Comment:
This makes a difference that previously we set flags == permissions = 0, which I believe were later enhanced with `rw-p` permissions throughout the `maintenance info sections` information.

And now, the `info_proc_maps` if it doesn't have permissions info, it sets `rwxp` and that's why our test fails.

So in the end we set wrong information even though we can have a more correct one here.

Reply:
I see, I'll add a parameter to `info_proc_maps` to only optionally set the flag.